lib/levenshtein.rb

```diff
@@ -4,10 +4,12 @@ module Levenshtein
   class << self
     extend FFI::Library
 
-    # Try loading in order.
-    library = File.dirname(__FILE__) + "/../ext/levenshtein/levenshtein"
-    candidates = ['.bundle', '.so', '.dylib', ''].map { |ext| library + ext }
-    ffi_lib(candidates)
+    dl_ext = RbConfig::MAKEFILE_CONFIG['DLEXT']
+    begin
+      ffi_lib File.expand_path("levenshtein.#{dl_ext}", __dir__)
+    rescue LoadError # Some environments don't copy a shared object over to lib
+      ffi_lib File.expand_path("levenshtein.#{dl_ext}", "#{__dir__}/../ext/levenshtein")
+    end
```

Comment:
This gem supports ruby versions that don't have `__dir__`. How about this:
```ruby
library = "levenshtein.#{RbConfig::MAKEFILE_CONFIG['DLEXT']}"
candidates = ["#{__FILE__}/..", "#{__FILE__}/../../ext/levenshtein"]
candidates.unshift(Gem.loaded_specs['levenshtein-ffi'].extension_dir) if defined?(Gem) && Gem.loaded_specs['levenshtein-ffi']
ffi_lib(candidates.map { |dir| File.expand_path(library, dir) })
```

Reply:
Thanks. Good call.